Library/Homebrew/cmd/list.rb

```diff
@@ -35,18 +50,18 @@ def list
       return
     end
 
-    if ARGV.include?("--pinned") || ARGV.include?("--versions")
-      filtered_list
+    if args.pinned? || args.versions?
+      filtered_list(args)
     elsif ARGV.named.empty?
-      if ARGV.include? "--full-name"
+      if args.full_name?
         full_names = Formula.installed.map(&:full_name).sort(&tap_and_name_comparison)
         return if full_names.empty?
         puts Formatter.columns(full_names)
       else
         ENV["CLICOLOR"] = nil
         exec "ls", *ARGV.options_only << HOMEBREW_CELLAR
       end
-    elsif ARGV.verbose? || !$stdout.tty?
+    elsif args.verbose? || !$stdout.tty?
```

Comment:
`Homebrew.args.verbose?`

Reply:
what is the purpose of adding `Homebrew.` ?

Reply:
Options like `verbose`, `force`, `quiet` and `debug` are project wide, hence they are stored as global vars to be accessible everywhere. 

If we have `--verbose` passed to a command and that calls method defined in some other file, it might also need `--verbose` and we don't want to be passing around them as parameters if possible.